awx/main/analytics/subsystem_metrics.py

```diff
@@ -179,9 +180,11 @@ def __init__(self, auto_pipe_execute=True, instance_name=None):
         # track last time metrics were sent to other nodes
         self.previous_send_metrics = SetFloatM('send_metrics_time', 'Timestamp of previous send_metrics call')
 
-    def clear_values(self):
-        for m in self.METRICS.values():
-            m.clear_value(self.conn)
+    def clear_values(self, fields=None):
+        if not fields:
+            fields = self.METRICS.keys()
```

Comment:
This seems a little weird. What's the reasoning behind this conditional? Might be worth a comment.

Reply:
I may or may not have stolen this from @fosterseth 's open PR 😅 
What I _believe_ is that it's resetting the values at the end of each partition in preparation for the next cycle.

Reply:
I think the best thing to do would be to revert the changes here before merge. As it stands, it's like adding mystery code.

Reply:
oh, I didn't realize it was fixing something real. I guess it's in the nuance of how it interacts with the python library?